di.go

```diff
@@ -349,9 +348,9 @@ func initializeInstance(beanID string, instance interface{}) error {
 	initializingBean := reflect.TypeOf((*InitializingBean)(nil)).Elem()
 	bean := reflect.TypeOf(instance)
 	if bean.Implements(initializingBean) {
-		initializingMethod, ok := bean.MethodByName(initializingBean.Method(0).Name)
+		initializingMethod, ok := bean.MethodByName("PostConstruct")
```

Comment:
Why hardcoding the method name here? `initializingBean` here is of interface `InitializingBean`, which has exactly one method `PostConstruct`, so if we address it as `initializingBean.Method(0).Name`, we will always get the correct method. And it doesn't depend on its name.

Reply:
Mostly this is kind of low-level optimization to avoid few calls, like .Method(0) and .Name.
Do you see any drawbacks of that?

> And it doesn't depend on its name.
The way i see it you are not gonna change it anymore (never) cause it will break backward compatibility.

Reply:
Maybe I'm wrong, but this will happen in compile-time, because Go _(almost)_ doesn't have a runtime. So it shouldn't really affect performance at all. And I'd just prefer to avoid string literals like this in the code...

Reply:
I have reverted code back to proceed with review.

How do you imagine it will execute it on compile time? I tried to find something about that, but no luck. 
Here is for example proposal for compilation time code and rejected...
https://github.com/golang/go/issues/39216
Here is list of compiler optimizations, looks like there is no such opt
https://github.com/golang/go/wiki/CompilerOptimizations

However there is a possibility to implement part of DI code in compile time with generate... it is possible to generate code and metadata info to avoid many different validations & checks in runtime, especially dependencies search
https://blog.golang.org/generate

I have created 2 benchmarks to test this. Benchmarks are simple, but i know that there is always some  internals that could behave differently in different conditions.
BenchmarkPostConstructNew-12    	169108303	         6.96 ns/op
BenchmarkPostConstructNew-12    	170593088	         7.02 ns/op
BenchmarkPostConstructNew-12    	167255487	         7.23 ns/op
BenchmarkPostConstructNew-12    	160335596	         7.38 ns/op
BenchmarkPostConstructNew-12    	169502205	         7.10 ns/op
BenchmarkPostConstructNew-12    	169132682	         7.10 ns/op

BenchmarkPostConstructPrevious-12    	158303479	         7.59 ns/op
BenchmarkPostConstructPrevious-12    	158439031	         7.51 ns/op
BenchmarkPostConstructPrevious-12    	159714298	         7.44 ns/op
BenchmarkPostConstructPrevious-12    	158489898	         7.49 ns/op
BenchmarkPostConstructPrevious-12    	159751230	         7.51 ns/op
BenchmarkPostConstructPrevious-12    	157850439	         7.55 ns/op

Here they are:
`type SomeBean struct {
}

func (fsb *SomeBean) PostConstruct() error {
	return nil
}
func BenchmarkPostConstructNew(b *testing.B) {
	typeB := reflect.TypeOf((*SomeBean)(nil))
	instance := reflect.New(typeB).Interface()
	initializingBean := reflect.TypeOf((*InitializingBean)(nil)).Elem()
	bean := reflect.TypeOf(instance)
	b.ResetTimer()
	for i := 0; i < b.N; i++ {
		if bean.Implements(initializingBean) {
			bean.MethodByName("PostConstruct")
		}
	}
}

func BenchmarkPostConstructPrevious(b *testing.B) {
	typeB := reflect.TypeOf((*SomeBean)(nil))
	instance := reflect.New(typeB).Interface()
	initializingBean := reflect.TypeOf((*InitializingBean)(nil)).Elem()
	bean := reflect.TypeOf(instance)
	b.ResetTimer()
	for i := 0; i < b.N; i++ {
		if bean.Implements(initializingBean) {
			bean.MethodByName(initializingBean.Method(0).Name)

		}
	}
}`